resources/assets/js/components/Application/BasicInfo/BasicInfoPage.tsx

```diff
@@ -1,59 +1,121 @@
-import fakeJob from "../../../fakeData/fakeJob";
+import { Job, ApplicationNormalized } from "../../../models/types";
+import { DispatchType } from "../../../configureStore";
+import { RootState } from "../../../store/store";
+import {
+  getApplication,
+  getApplicationIsUpdating,
+} from "../../../store/Application/applicationSelector";
+import {
+  updateApplication as updateApplicationAction,
+  fetchApplicationNormalized,
+} from "../../../store/Application/applicationActions";
+import { fetchJob } from "../../../store/Job/jobActions";
+import { getJob, getJobIsUpdating } from "../../../store/Job/jobSelector";
 
 interface BasicInfoPageProps {
   applicationId: number;
 }
 
-export const BasicInfoPage: React.FunctionComponent<BasicInfoPageProps> = ({
+const BasicInfoPage: React.FunctionComponent<BasicInfoPageProps> = ({
   applicationId,
 }) => {
   const intl = useIntl();
   const locale = getLocale(intl.locale);
+  const dispatch = useDispatch<DispatchType>();
 
-  const application = fakeApplication(); // TODO: get real application.
-  const job = fakeJob(); // TODO: Get real job associated with application.
+  // Load Application.
+  const applicationSelector = (
+    state: RootState,
+  ): ApplicationNormalized | null => getApplication(state, { applicationId });
+  const application = useSelector(applicationSelector);
+  const applicationIsUpdating = useSelector((state: RootState) =>
+    getApplicationIsUpdating(state, { applicationId }),
+  );
 
-  const handleContinue = (values): void => {
-    // TODO: Save basic info form values.
+  useEffect(() => {
+    if (application === null && !applicationIsUpdating) {
+      dispatch(fetchApplicationNormalized(applicationId));
+    }
+  }, [dispatch, applicationId, application, applicationIsUpdating]);
+
+  // Load Job.
+  const jobId = application?.job_poster_id;
+  const jobSelector = (state: RootState): Job | null =>
+    jobId ? getJob(state, { jobId }) : null;
+  const job = useSelector(jobSelector);
+  const jobUpdatingSelector = (state: RootState): boolean =>
+    jobId ? getJobIsUpdating(state, jobId) : false;
+  const jobIsUpdating = useSelector(jobUpdatingSelector);
+
+  useEffect(() => {
+    // If job is null and not already updating, fetch it.
+    if (jobId && job === null && !jobIsUpdating) {
+      dispatch(fetchJob(jobId));
+    }
+  }, [jobId, job, jobIsUpdating, dispatch]);
+
+  if (application === null || job === null) {
+    return null;
+  }
+
+  const updateApplication = async (
+    editedApplication: ApplicationNormalized,
+  ): Promise<ApplicationNormalized> => {
+    const result = await dispatch(updateApplicationAction(editedApplication));
+    if (!result.error) {
+      const payload = await result.payload;
+      return payload;
+    }
+    return Promise.reject(result.payload);
+  };
+
+  const handleContinue = (values: ApplicationNormalized): void => {
+    updateApplication(values);
     navigate(applicationExperienceIntro(locale, applicationId));
```

Comment:
As is, this will start the update but navigate away before it completes (and even if it fails). Is that what we want?